cdc/http_handler.go

```diff
@@ -260,6 +280,100 @@ func (s *Server) handleChangefeedQuery(w http.ResponseWriter, req *http.Request)
 	writeData(w, resp)
 }
 
+// handleChangefeeds dispatch the request to the specified handleFunc according to the request method.
+func (s *Server) handleChangefeeds(w http.ResponseWriter, req *http.Request) {
+	if req.Method == http.MethodGet {
+		s.handleChangefeedsList(w, req)
+		return
+	}
+}
+
+func (s *Server) handleChangefeedsList(w http.ResponseWriter, req *http.Request) {
+	if req.Method != http.MethodGet {
+		writeError(w, http.StatusBadRequest, cerror.ErrSupportPostOnly.GenWithStackByArgs())
+		return
+	}
+	s.ownerLock.RLock()
+	defer s.ownerLock.RUnlock()
+	if s.owner == nil {
+		handleOwnerRespJSON(w, concurrency.ErrElectionNotLeader)
+		return
+	}
+
+	err := req.ParseForm()
+	if err != nil {
+		writeInternalServerErrorJSON(w, cerror.WrapError(cerror.ErrInternalServerError, err))
+		return
+	}
+
+	statuses, err := s.owner.etcdClient.GetAllChangeFeedStatus(req.Context())
+	changefeedIDs := make(map[string]struct{}, len(statuses))
+	if err != nil {
+		writeInternalServerErrorJSON(w, err)
+		return
+	}
+	for cid := range statuses {
+		changefeedIDs[cid] = struct{}{}
+	}
+
+	state := req.Form.Get(APIOpVarChangefeeds)
+	filters := map[model.FeedState]struct{}{
```

Comment:
Can you add comments on the meaning of `filters`? It seems that whatever is in `filters` gets filtered out. Probably the other way around is clearer, i.e., filtering out whatever is NOT in `filters`.

Reply:
done